resources/aws/subnet.go

```diff
@@ -0,0 +1,116 @@
+package aws
+
+import (
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	microerror "github.com/giantswarm/microkit/error"
+
+	awsclient "github.com/giantswarm/aws-operator/client/aws"
+)
+
+type Subnet struct {
+	AvailabilityZone string
+	CidrBlock        string
+	VpcID            string
+	id               string
+	AWSEntity
+}
+
+func (s *Subnet) checkIfExists() (bool, error) {
```

Comment:
Exaclty, we cannot "template" that function in any way, so unfortunatelly, every `checkIfExists` and `CreateIfNotExists` functions will look very similar.